import-service/shared/utils/s3Utils.ts

```diff
@@ -0,0 +1,32 @@
+import S3 from "aws-sdk/clients/s3";
+import csvParser from 'csv-parser';
+
+const s3 = new S3({
+    region: process?.env?.PRODUCT_AWS_REGION || 'eu-north-1',
+})
+
+export const parseS3CSVObjectStream = async (s3ObjectStream: any, onEndCallback: any): Promise<void> => {
+    await new Promise((res, rej) => {
+        s3ObjectStream
+            .pipe(csvParser())
+            .on('data', (chunk: string) => {
+                console.log('File chink data: ', chunk)
+            })
+            .on('end', async () => {
+                await onEndCallback()
```

Comment:
No need to use callbacks while using promise, one can just wait for the promise to resolve and continue further operations (copy/delete s3 object in your case).
Moreover in the current implementation promise won't be resolved until copy/delete operations finish.